p2p/protocol/circuitv1/relay/relay.go

```diff
@@ -108,55 +136,74 @@ func (r *Relay) handleStream(s network.Stream) {
 }
 
 func (r *Relay) handleHopStream(s network.Stream, msg *pb.CircuitRelay) {
+	txn, err := r.scope.BeginTransaction()
```

Comment:
Are we sure that "transaction" is the right term here? To me, a transaction denotes logical atomicity and isolation of some kind, which I'm not sure is the right semantics here. There is no atomicity (succeed or fail all) nor isolation (reservations are in effect immediately before the transaction is done). Looks like this thing is just a convenience API to model lifetimes/spans, right? In that case, would `Block` or `Span` fit better as a concept?

Reply:
yeah, that's right. Maybe `Block/Span` is a better term indeed.

Reply:
changed to Span.